src/modules/networks/Networks.ts

```diff
@@ -25,152 +25,75 @@ function toCollection<T, K>(
-  private originalCollection: { [key: string]: NetworkConfig | undefined };
-  private supportedByBackend: Set<string>;
+  private ethereumChainConfigs: EthereumChainConfig[];
   static purposeKeyMap = {
     sending: 'supports_sending',
     trading: 'supports_trading',
-    bridge: 'supports_bridge',
+    bridge: 'supports_bridging',
+    actions: 'supports_actions',
+    nfts: 'supports_nft_positions',
+    positions: 'supports_positions',
   } as const;
 
   constructor({
     networks,
-    ethereumChainSources,
+    ethereumChainConfigs,
   }: {
     networks: NetworkConfig[];
-    ethereumChainSources?: EthereumChainSources;
+    ethereumChainConfigs: EthereumChainConfig[];
   }) {
-    this.networks = networks
-      .filter((network) => Boolean(network.evm_id))
-      .sort((a, b) => localeCompareWithPriority(a.name, b.name, 'Ethereum'));
-    this.ethereumChainSources = ethereumChainSources;
-    const {
-      collection,
-      originalCollection,
-      collectionByEvmId,
-      sourcesNormalized,
-    } = this.prepare();
-    this.collection = collection;
-    this.originalCollection = originalCollection;
-    this.collectionByEvmId = collectionByEvmId;
-    this.sourcesNormalized = sourcesNormalized;
-    this.supportedByBackend = new Set(networks.map((n) => n.chain));
-  }
-
-  private prepare() {
-    const originalCollection = toCollection(
+    this.ethereumChainConfigs = ethereumChainConfigs;
+    this.networks = insertChainConfigs(networks, ethereumChainConfigs);
+    this.collection = toCollection(
       this.networks,
-      (network) => network.chain,
+      (network) => network.id,
       (x) => x
     );
-    const collectionByEvmId = toCollection(
+    this.collectionByEvmId = toCollection(
       this.networks,
-      (x) => x.external_id,
+      (x) => getChainId(x) || '',
       (x) => x
-    );
-    const sourcesNormalized = normalizeSources(this.ethereumChainSources);
-    const collection = { ...originalCollection };
-    for (const value of Object.values(sourcesNormalized)) {
-      if (!value) {
-        continue;
-      }
-      Object.assign(collection, value.collection);
-      Object.assign(
-        collectionByEvmId,
-        toCollection(
-          value.items,
-          (item) => item.external_id,
-          (item) => item
-        )
-      );
-    }
-    return {
-      collection,
-      originalCollection,
-      collectionByEvmId,
-      sourcesNormalized,
-    };
-  }
-
-  updateEthereumChainSources(ethereumChainSources: EthereumChainSources) {
-    this.ethereumChainSources = ethereumChainSources;
-    const {
-      collection,
-      collectionByEvmId,
-      originalCollection,
-      sourcesNormalized,
-    } = this.prepare();
-    this.collection = collection;
-    this.originalCollection = originalCollection;
-    this.collectionByEvmId = collectionByEvmId;
-    this.sourcesNormalized = sourcesNormalized;
+    ) as { [key: number]: NetworkConfig | undefined };
```

Comment:
why is `key: number`? `getChainId` seems to return a string, why do we have this casting?

Reply:
Good catch!